pipeline/runtime/execer.go

```diff
@@ -232,24 +236,36 @@ func (e *Execer) exec(ctx context.Context, state *pipeline.State, spec Spec, ste
 	wc := e.streamer.Stream(noContext, state, step.GetName())
 	wc = newReplacer(wc, secretSlice(step))
 
+	// wrap writer in extrator
+	ext := extractor.New(wc)
+
 	// if the step is configured as a daemon, it is detached
 	// from the main process and executed separately.
 	if step.IsDetached() {
 		go func() {
-			e.engine.Run(ctx, spec, copy, wc)
+			e.engine.Run(ctx, spec, copy, ext)
 			wc.Close()
 		}()
 		return nil
 	}
 
-	exited, err := e.engine.Run(ctx, spec, copy, wc)
+	exited, err := e.engine.Run(ctx, spec, copy, ext)
 
 	// close the stream. If the session is a remote session, the
 	// full log buffer is uploaded to the remote server.
 	if err := wc.Close(); err != nil {
 		result = multierror.Append(result, err)
 	}
 
+	// upload card if exists
+	card, ok := ext.File()
+	if ok {
+		err = e.uploader.UploadCard(ctx, card, state, step.GetName())
+		if err != nil {
+			return nil
```

Comment:
If we receive an error uploading the card we probably want to collect the error and then continue (similar to how we handle an error in wc.Close above in line 257).

```
	if err != nil {
		log.Warnln("cannot upload card")
		result = multierror.Append(result, err)
	}
```

Reply:
ah my bad, i forgot to update this.